### ISSUE TITLE
What are the specs for the various on-boarding scenarios?

### ISSUE THREAD

**Issue by CodySwannGT** [gunnertech/thumbwar-ios-client#8]
Currently, there is a hole in the specs:

Right now, when you create a ThumbWar, the "who" field populates with three forms of data:

1) Your Thumbwar Friends
2) People from your Address Book
3) Your Facebook friends who are on Facebook

The first scenario is clear what happens.

But what happens when we select someone from the other two scenarios?

**Comment by CodySwannGT** [comment]
@smummolo - got anything?

**Comment by smummolo** [comment]
im not sure what the question is - this seems to be working fine?

**Comment by adamespn** [comment]
Based on #48, I believe we have only people from categories 2 and 3. Can you confirm?

**Comment by jiakeliu** [comment]
Is category 1, Thumbwar friends, currently using the system generated username (firstName+lastName+num)?

**Comment by adamespn** [comment]
It looks like username is the field being looked up and Fn/ln is being displayed

**Comment by adamespn** [comment]
should this be closed?

**Comment by jiakeliu** [comment]
Can we combine "TW: First Last" "FB: First Last" and "AB: First Last" to just display "First Last"?

**Comment by adamespn** [comment]
They could be different people - better to just link in the table cells from user search later - it will be nicer to look at and self-explanatory

**Comment by jiakeliu** [comment]
What do you mean by they can be different people? To a user, it doesn't
really concern them whether they are challenging a friend via Address Book,
Facebook, or TW, as long as they can challenge that friend with a given
name.

The table cells from user search only list the name and not the sources
(FB, AB, and TW) right?

**Comment by adamespn** [comment]
I can have a contact named Rob smith and a Facebook friend named rob smith who is different and I can be following yet a third Rob smith in TW.

User search treats them all as different people as well.

We have discussed user resolution in the past but no good way to do it was proposed.

**Comment by jiakeliu** [comment]
The search feed has a small profile picture to the left right? That will
help distinguish people with the same name.

Let's also display the unique username next to the full name, so the end
result should be like this:

*[PIC] Adam Meikle* @adamespn
*[PIC] Adam Meikle* @anotheradam

The thing we for sure do not want is this:

FB: *Adam Meikle* @adamespn
AB: *Adam Meikle* @adamespn
TW: *Adam Meikle* @adamespn

Does that make sense?

**Comment by jiakeliu** [comment]
Here's how Twitter handles it:

https://twitter.com/search?q=bob%20smith&mode=users

**Comment by adamespn** [comment]
They don't have usernames unless they are TW users

**Comment by CodySwannGT** [comment]
That's actually not true - all users upon creation are given a username.

Unless I'm misreading the context here, but at the database level, all user
records have a username

On Wed, Apr 29, 2015 at 10:29 PM, adamespn <notifications@github.com> wrote:

> They don't have usernames unless they are TW users
>
>
>
>
> > On Apr 29, 2015, at 10:14 PM, Jiake Liu <notifications@github.com>
> wrote:
> >
> > The search feed has a small profile picture to the left right? That will
> > help distinguish people with the same name.
> >
> > Let's also display the unique username next to the full name, so the end
> > result should be like this:
> >
> > *[PIC] Adam Meikle* @adamespn
> > *[PIC] Adam Meikle* @anotheradam
> >
> > The thing we for sure do not want is this:
> >
> > FB: *Adam Meikle* @adamespn
> > AB: *Adam Meikle* @adamespn
> > TW: *Adam Meikle* @adamespn
> >
> > Does that make sense?
> >
> > On Wed, Apr 29, 2015 at 6:46 PM, adamespn <notifications@github.com>
> wrote:
> >
> > > I can have a contact named Rob smith and a Facebook friend named rob
> smith
> > > who is different and I can be following yet a third Rob smith in TW.
> > >
> > > User search treats them all as different people as well.
> > >
> > > We have discussed user resolution in the past but no good way to do it
> was
> > > proposed.
> > >
> > >
> > >
> > >
> > >
> > > > On Apr 29, 2015, at 9:43 PM, Jiake Liu <notifications@github.com>
> wrote:
> > > >
> > > > What do you mean by they can be different people? To a user, it
> doesn't
> > > > really concern them whether they are challenging a friend via Address
> > > Book,
> > > > Facebook, or TW, as long as they can challenge that friend with a
> given
> > > > name.
> > > >
> > > > The table cells from user search only list the name and not the
> sources
> > > > (FB, AB, and TW) right?
> > > > On Apr 29, 2015 6:29 PM, "adamespn" <notifications@github.com>
> wrote:
> > > >
> > > > > They could be different people - better to just link in the table
> cells
> > > > > from user search later - it will be nicer to look at and
> > > self-explanatory
> > > > >
> > > > >
> > > > >
> > > > > > On Apr 29, 2015, at 3:28 PM, Jiake Liu <notifications@github.com
> >
> > > wrote:
> > > > > >
> > > > > > Can we combine "TW: First Last" "FB: First Last" and "AB: First
> > > Last" to
> > > > > just display "First Last"?
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub.
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > >
> https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97628859
> > > >
> > >
> > > > > .
> > > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub.
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97630344
> >
> > > .
> > >
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97636499>
> .
>



-- 
 Cody Swann - CEO, Gunner Technology
T: 213.915.4083 | cody@gunnertech.com | www.gunnertech.com
<http://www.gunnertech.com/>Skype: cody.swann

*CONFIDENTIALITY NOTICE: This email message, together with any documents,
files and/or email messages attached to it, is intended for the sole use of
the individual or entity to whom it is addressed, and may contain
information that is legally privileged, confidential and restricted from
disclosure.  If you are not the intended recipient, or responsible for
delivery to that person, you are hereby notified that any dissemination or
copying of this communication is STRICTLY PROHIBITED.  In such case please
notify the sender by reply email and delete this message without reading,
printing or saving.*

**Comment by adamespn** [comment]
People in your address book or on Facebook that are not users do not have a username

**Comment by CodySwannGT** [comment]
Oh gotcha. Right. That's correct.

On Wed, Apr 29, 2015 at 10:39 PM, adamespn <notifications@github.com> wrote:

> People in your address book or on Facebook that are not users do not have
> a username
>
>
>
>
> > On Apr 29, 2015, at 10:37 PM, Cody Swann <notifications@github.com>
> wrote:
> >
> > That's actually not true - all users upon creation are given a username.
> >
> > Unless I'm misreading the context here, but at the database level, all
> user
> > records have a username
> >
> > On Wed, Apr 29, 2015 at 10:29 PM, adamespn <notifications@github.com>
> wrote:
> >
> > > They don't have usernames unless they are TW users
> > >
> > >
> > >
> > >
> > > > On Apr 29, 2015, at 10:14 PM, Jiake Liu <notifications@github.com>
> > > wrote:
> > > >
> > > > The search feed has a small profile picture to the left right? That
> will
> > > > help distinguish people with the same name.
> > > >
> > > > Let's also display the unique username next to the full name, so the
> end
> > > > result should be like this:
> > > >
> > > > *[PIC] Adam Meikle* @adamespn
> > > > *[PIC] Adam Meikle* @anotheradam
> > > >
> > > > The thing we for sure do not want is this:
> > > >
> > > > FB: *Adam Meikle* @adamespn
> > > > AB: *Adam Meikle* @adamespn
> > > > TW: *Adam Meikle* @adamespn
> > > >
> > > > Does that make sense?
> > > >
> > > > On Wed, Apr 29, 2015 at 6:46 PM, adamespn <notifications@github.com>
> > > wrote:
> > > >
> > > > > I can have a contact named Rob smith and a Facebook friend named
> rob
> > > smith
> > > > > who is different and I can be following yet a third Rob smith in
> TW.
> > > > >
> > > > > User search treats them all as different people as well.
> > > > >
> > > > > We have discussed user resolution in the past but no good way to
> do it
> > > was
> > > > > proposed.
> > > > >
> > > > >
> > > > >
> > > > >
> > > > >
> > > > > > On Apr 29, 2015, at 9:43 PM, Jiake Liu <notifications@github.com
> >
> > > wrote:
> > > > > >
> > > > > > What do you mean by they can be different people? To a user, it
> > > doesn't
> > > > > > really concern them whether they are challenging a friend via
> Address
> > > > > Book,
> > > > > > Facebook, or TW, as long as they can challenge that friend with a
> > > given
> > > > > > name.
> > > > > >
> > > > > > The table cells from user search only list the name and not the
> > > sources
> > > > > > (FB, AB, and TW) right?
> > > > > > On Apr 29, 2015 6:29 PM, "adamespn" <notifications@github.com>
> > > wrote:
> > > > > >
> > > > > > > They could be different people - better to just link in the
> table
> > > cells
> > > > > > > from user search later - it will be nicer to look at and
> > > > > self-explanatory
> > > > > > >
> > > > > > >
> > > > > > >
> > > > > > > > On Apr 29, 2015, at 3:28 PM, Jiake Liu <
> notifications@github.com
> > > >
> > > > > wrote:
> > > > > > > >
> > > > > > > > Can we combine "TW: First Last" "FB: First Last" and "AB:
> First
> > > > > Last" to
> > > > > > > just display "First Last"?
> > > > > > > >
> > > > > > > > —
> > > > > > > > Reply to this email directly or view it on GitHub.
> > > > > > > >
> > > > > > >
> > > > > > > —
> > > > > > > Reply to this email directly or view it on GitHub
> > > > > > > <
> > > > >
> > >
> https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97628859
> > > > > >
> > > > >
> > > > > > > .
> > > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub.
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > >
> https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97630344
> > > >
> > > > > .
> > > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub.
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97636499
> >
> > > .
> > >
> >
> >
> >
> > --
> > Cody Swann - CEO, Gunner Technology
> > T: 213.915.4083 | cody@gunnertech.com | www.gunnertech.com
> > <http://www.gunnertech.com/>Skype: cody.swann
> >
> > *CONFIDENTIALITY NOTICE: This email message, together with any documents,
> > files and/or email messages attached to it, is intended for the sole use
> of
> > the individual or entity to whom it is addressed, and may contain
> > information that is legally privileged, confidential and restricted from
> > disclosure. If you are not the intended recipient, or responsible for
> > delivery to that person, you are hereby notified that any dissemination
> or
> > copying of this communication is STRICTLY PROHIBITED. In such case please
> > notify the sender by reply email and delete this message without reading,
> > printing or saving.*
>
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97637837>
> .
>



-- 
 Cody Swann - CEO, Gunner Technology
T: 213.915.4083 | cody@gunnertech.com | www.gunnertech.com
<http://www.gunnertech.com/>Skype: cody.swann

*CONFIDENTIALITY NOTICE: This email message, together with any documents,
files and/or email messages attached to it, is intended for the sole use of
the individual or entity to whom it is addressed, and may contain
information that is legally privileged, confidential and restricted from
disclosure.  If you are not the intended recipient, or responsible for
delivery to that person, you are hereby notified that any dissemination or
copying of this communication is STRICTLY PROHIBITED.  In such case please
notify the sender by reply email and delete this message without reading,
printing or saving.*

**Comment by jiakeliu** [comment]
Got it.

Unlike the address book, I don't think Facebook allows searching for users that have not authorized an app to access their profile, and the only way a user can authorize Thumbwar to access their profile is by signing up on Thumbwar. Therefore you technically can't find Facebook friends that are not already a Thumbwar user, correct?

So I think the correct UX design here is to take out the "FB: Adam Meikle" row and keep "TW: Adam Meikle" and "AB: Adam Meikle" The UI also needs to be changed, because new users wouldn't understand what "TW" or "AB" is.

In summary, let's make these changes:

"FB: Adam Meikle" ----> take this row out
"AB: Adam Meikle" ----> "[Phone Logo] Adam Meikle (203) 273-2111"
"TW: Adam Meikle" ----> "[Profile Pic] Adam Meikle @adamespn"

**Comment by adamespn** [comment]
We know who all your Facebook friends are, so that assumption is incorrect.

**Comment by jiakeliu** [comment]
What can we do with a Facebook friend that shows up on Thumbwar search that's not already a Thumbwar user? Can we send them a Facebook message?

**Comment by CodySwannGT** [comment]
No. All you can really do is use it as a reminder.

Like, the way I would do it is when you click a Facebook user (who isn't
already on Thumbwar) it prompts you for a phone number or email to send the
invitation to.

Or just leave out Facebook users who aren't connected to Thumbwar

On Thu, Apr 30, 2015 at 3:37 PM, Jiake Liu <notifications@github.com> wrote:

> What can we do with a Facebook friend that shows up on Thumbwar search
> that's not already a Thumbwar user? Can we send them a Facebook message?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gunnertech/thumbwar-ios-client/issues/8#issuecomment-97940493>
> .
>



-- 
 Cody Swann - CEO, Gunner Technology
T: 213.915.4083 | cody@gunnertech.com | www.gunnertech.com
<http://www.gunnertech.com/>Skype: cody.swann

*CONFIDENTIALITY NOTICE: This email message, together with any documents,
files and/or email messages attached to it, is intended for the sole use of
the individual or entity to whom it is addressed, and may contain
information that is legally privileged, confidential and restricted from
disclosure.  If you are not the intended recipient, or responsible for
delivery to that person, you are hereby notified that any dissemination or
copying of this communication is STRICTLY PROHIBITED.  In such case please
notify the sender by reply email and delete this message without reading,
printing or saving.*

**Comment by adamespn** [comment]
The app currently supports starting a war with a Facebook friend who is not currently a thumbwar user as well as a contact who is not currently a Thumbwar user

**Comment by jiakeliu** [comment]
Adam, how do you currently start a war with a Facebook friend who is not a thumbwar user? A contact I understand is through SMS.

**Comment by adamespn** [comment]
You type "Jiake Liu" and their name comes up as a Facebook friend as "FB: Jiake Liu".
You select them. You start a war. TW app creates an initial record for the user in TW. The war is created.
I do not know if the server then creates a Facebook notification of some sort.

**Comment by jiakeliu** [comment]
If John Smith is not a TW user, but he's a friend of mine on FB, then even if I create a war like the way you described FB doesn't allow apps to send notifications on their platform. So for now, let's just take out the FW row and stick to the screenshot above.

**Comment by CodySwannGT** [comment]
On the call we decided:

1) No entity resolution

2) Load only Facebook users who have authorized ThumbWar

**Comment by husseinm** [comment]
Wasn't this implemented?

**Comment by jiakeliu** [comment]
The icons (portrait and the phone icon) weren't implemented.

**Comment by jiakeliu** [comment]
I'm still seeing "AB: Mahdi Hussein" instead of "[Phone Icon] Mahdi Hussein" in the user lookup in new bet screen.

**Comment by jiakeliu** [comment]
@husseinm Can we cram this in this iteration?
